### PR TITLE
Update kustomization.yaml files to new format

### PR DESCRIPTION
**Pull request by thunderboltsid**
kustomize has deprecated the older format which our older kustomization files were based on. These older format files were leading to a lot of deprecation warnings from kustomize during our build processes.

We used `kustomize edit fix` command to update our old kustomize files to our new kustomize files. This PR was largely auto-generated using the following command.

```
$ find . -type d -exec bash -c 'cd {} && kustomize edit fix' \;
```

### CODE REVIEW

**Comment by deepakm-ntnx** [comment]
pls hold this change till we merge https://github.com/d2iq-labs/capi-runtime-extensions/pull/384 as these are easy to make changes than the one done in https://github.com/d2iq-labs/capi-runtime-extensions/pull/384

**Comment by thunderboltsid** [comment]
> pls hold this change till we merge [d2iq-labs/capi-runtime-extensions#384](https://github.com/d2iq-labs/capi-runtime-extensions/pull/384) as these are easy to make changes than the one done in [d2iq-labs/capi-runtime-extensions#384](https://github.com/d2iq-labs/capi-runtime-extensions/pull/384)

This changes the format of kustomization.yaml files to the ones expected by kustomize. This is pretty much an auto-generated change from kustomize linter. Can you clarify why the other PR is a blocker for this PR?

**Comment by deepakm-ntnx** [comment]
> > pls hold this change till we merge [d2iq-labs/capi-runtime-extensions#384](https://github.com/d2iq-labs/capi-runtime-extensions/pull/384) as these are easy to make changes than the one done in [d2iq-labs/capi-runtime-extensions#384](https://github.com/d2iq-labs/capi-runtime-extensions/pull/384)
> 
> This changes the format of kustomization.yaml files to the ones expected by kustomize. This is pretty much an auto-generated change from kustomize linter. Can you clarify why the other PR is a blocker for this PR?

https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/386

**Comment by thunderboltsid** [comment]
That makes sense. Given these changes are auto-generated anyway, it's easy enough to run the command again once the other PR is merged.

**Review by deepakm-ntnx — APPROVED** [review]

**Review by yannickstruyf3 — APPROVED** [review]

**Pull request merged**
